dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/SampleToBatchPadding.scala

```diff
@@ -0,0 +1,184 @@
+ fixLength: Option[Int] = None)
+(implicit ev: TensorNumeric[T])
+  extends Transformer[Sample[T], MiniBatch[T]] {
+
+  private def padding(data: Array[T], padValue: T): Unit = {
+    ev.getType() match {
+      case DoubleType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Double]], 0, data.length,
+          ev.toType[Double](padValue))
+      case FloatType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Float]], 0, data.length,
+          ev.toType[Float](padValue))
+      case _ => throw new UnsupportedOperationException(
+        "SampleToBatchPadding: Only Float/Double supported")
+    }
+  }
+
+  override def apply(prev: Iterator[Sample[T]]): Iterator[MiniBatch[T]] = {
+    new Iterator[MiniBatch[T]] {
+      private val featureTensor: Tensor[T] = Tensor[T]()
+      private val labelTensor: Tensor[T] = Tensor[T]()
+      private var featureData: Array[T] = null
+      private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
+      private val batchSize = Utils.getBatchSize(totalBatch)
+      private var featureSize: Array[Int] = null
+      private var labelSize: Array[Int] = null
+      private var oneFeatureLength: Int = 0
+      private var oneLabelLength: Int = 0
+      private var padFeatureLength: Int = 0
+      private var padLabelLength: Int = 0
+      override def hasNext: Boolean = prev.hasNext
+
+      override def next(): MiniBatch[T] = {
+        if (prev.hasNext) {
+          var i = 0
+          var maxfeatureElement = 0
+          var maxlabelElement = 0
+          var labelIndex = 0
+          var featureIndex = 0
+          var padLength: Int = 0
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
+          while (i < batchSize && prev.hasNext) {
+            val sample = prev.next()
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
+            if ((maxfeatureElement != 0) && (!padFeature)) {
+              require(sample.feature().nElement() == maxfeatureElement,
+                "SampleToBatchPadding: all feature size should be same when not padding feature")
+            }
+            if ((maxlabelElement != 0) && (!padLabel)) {
+              require(sample.label().nElement() == maxlabelElement,
+                "SampleToBatchPadding: all label size should be same when not padding label")
+            }
+            sampleData(i).copy(sample)
+            if (sample.feature().nElement() > maxfeatureElement) {
+              maxfeatureElement = sample.feature().nElement()
+              featureIndex = i
+            }
+            if (sample.label().nElement() > maxlabelElement) {
+              maxlabelElement = sample.label().nElement()
+              labelIndex = i
+            }
```

Comment:
Since it is the same operation for feature and label, how about wrapping the operation to a private function and call that function both for feature and label?

private def less(int a, int b, int i, int j) = if (a < b) (b, j) else (a, i)
maxfeatureElement = less(sample.feature().nElement, maxfeatureElement, i, featureIndex)
maxlabelElement = less(sample.label().nElement, maxlabelElement, i, labelIndex)